src/actors.rs

```diff
@@ -298,7 +298,7 @@ where
             .collect();
         let difference: HashSet<u32> = missing_storage_nums
             .difference(&blocks)
-            .map(|b| *b)
+            .copied()
```

Comment:
This change is good. I wonder however if we couldn't combine `missing_storage_nums` and this pile of combinators into one, thus avoiding having to allocate for `missing_storage_nums`?

Reply:
Ok, I just combined it into one, however I don't see how this avoids a non-constant allocation. From my understanding we would still have to allocate at `.collect::<HashSet<u32>>()`, which is the equivalent of `missing_storage_nums` minus the variable binding on the stack. But I am guessing I am missing something?

Reply:
I don't know how what you have now optimizes, but before we had two `collect()` calls (which each allocates); with this last change I'd hope we only allocate once and for a smaller collection, i.e. just the difference.